File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    std::vector<int> v;
    string str = to_string(n);
    for (int i = 0; i < str.size(); i++) {
        v.emplace_back(str[i]-'0');
    }
    sort(v.begin(), v.end(), greater<>());
    string ans = "";
    for (int i = 0; i < v.size(); i++) {
        ans += to_string(v[i]);
    }
    answer = stoll(ans);
    return answer;
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
import java.util.*;

class Solution {
    public long solution(long n) {
        long answer = 0;
        String s = String.valueOf(n);
        String S = "";
        // ArrayList<String> list = new ArrayList<String>();
        // for (int i = 0; i < s.length(); i++) {
        //     char c = s.charAt(i);
        //     list.add(String.valueOf(c));
        // }
        ArrayList<String> list = new ArrayList(Arrays.asList(s.split("")));
        list.sort(Comparator.reverseOrder());
        // for (int i = 0; i < list.size(); i++) {
        //     S = S + list.get(i);
        // }
        StringBuilder sb = new StringBuilder();
        for (String l : list) {
            sb.append(l);
        }
        // answer = Long.valueOf(S);
        answer = Long.valueOf(sb.toString());
        return answer;
    }
}